### PR TITLE
chore: update files and folder names

### LINKED ISSUE #5: Directory Structure
**What is wrong?**
`www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection` is too repetitive.

**What is a solution?**
Update the structure to something like the following:
`www-project-top-ten/2017/A1-Injection`

**How could it be done?**
Update the file names to exactly what they are, `AX-<what it is>`
Update the folder name to follow the above mentioned nomenclature.

### PR DESCRIPTION
**Pull request by ThunderSon**
This PR closes #5 

Commands used:
`rename 's/Top.*2017_(.*)/$1/' *.md`
`mv OWASP_Top_Ten_2017/ 2017`

@hblankenship the only concern would be redirection if there is any that is occurring. Let me know what you think about the new naming convention

### CODE REVIEW

**Comment by ThunderSon** [comment]
@sslHello This should take into consideration what @kingthorin proposed in #5 
I updated the ToC as well to follow the simple naming convention.

**Comment by kingthorin** [comment]
Will test locally in a few hours.

**Comment by kingthorin** [comment]
Tested, found more that needed tweaking:

```diff
diff --git a/2017/A1-Injection.md b/2017/A1-Injection.md
index d93b775..4368c54 100644
--- a/2017/A1-Injection.md
+++ b/2017/A1-Injection.md
@@ -13,7 +13,9 @@ exploitability: 3
 prevalence: 2
 detectability: 3
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A1-Injection.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A1-Injection.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A1-Injection
 
 ---
 
@@ -76,4 +78,4 @@ This changes the meaning of both queries to return all the records from the acco
 * [CWE-564: Hibernate Injection](https://cwe.mitre.org/data/definitions/564.html)<br>
 * [CWE-917: Expression Language Injection](https://cwe.mitre.org/data/definitions/917.html)<br>
 * [PortSwigger: Server-side template injection](https://portswigger.net/kb/issues/00101080_serversidetemplateinjection)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A10-Insufficient_Logging%2526Monitoring.md b/2017/A10-Insufficient_Logging%2526Monitoring.md
index c6e6901..d780c06 100644
--- a/2017/A10-Insufficient_Logging%2526Monitoring.md
+++ b/2017/A10-Insufficient_Logging%2526Monitoring.md
@@ -13,7 +13,9 @@ exploitability: 2
 prevalence: 3
 detectability: 1
 technical: 2
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A10-Insufficient_Logging%2526Monitoring.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A10-Insufficient_Logging%2526Monitoring.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A10-Insufficient_Logging%2526Monitoring
 
 ---
 
@@ -69,4 +71,4 @@ There are commercial and open source application protection frameworks such as [
 **External**<br>
 * [CWE-223: Omission of Security-relevant Information](https://cwe.mitre.org/data/definitions/223.html)<br>
 * [CWE-778: Insufficient Logging](https://cwe.mitre.org/data/definitions/778.html)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A2-Broken_Authentication.md b/2017/A2-Broken_Authentication.md
index 1053687..87d4d11 100644
--- a/2017/A2-Broken_Authentication.md
+++ b/2017/A2-Broken_Authentication.md
@@ -13,7 +13,9 @@ exploitability: 3
 prevalence: 2
 detectability: 2
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication
 
 ---
 
@@ -72,4 +74,4 @@ Confirmation of the user's identity, authentication, and session management are
 * [NIST 800-63b: 5.1.1 Memorized Secrets](https://pages.nist.gov/800-63-3/sp800-63b.html#memsecret)<br>
 * [CWE-287: Improper Authentication](https://cwe.mitre.org/data/definitions/287.html)<br>
 * [CWE-384: Session Fixation](https://cwe.mitre.org/data/definitions/384.html)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A3-Sensitive_Data_Exposure.md b/2017/A3-Sensitive_Data_Exposure.md
index c4e4649..f6ef117 100644
--- a/2017/A3-Sensitive_Data_Exposure.md
+++ b/2017/A3-Sensitive_Data_Exposure.md
@@ -13,7 +13,9 @@ exploitability: 2
 prevalence: 3
 detectability: 2
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A3-Sensitive_Data_Exposure.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A3-Sensitive_Data_Exposure.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A3-Sensitive_Data_Exposure
 
 ---
 
@@ -75,4 +77,4 @@ Do the following, at a minimum, and consult the references:<br>
 * [CWE-326: Weak Encryption](https://cwe.mitre.org/data/definitions/326.html)<br>
 * [CWE-327: Broken/Risky Crypto](https://cwe.mitre.org/data/definitions/327.html)<br>
 * [CWE-359: Exposure of Private Information (Privacy Violation)](https://cwe.mitre.org/data/definitions/359.html)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A4-XML_External_Entities_(XXE).md b/2017/A4-XML_External_Entities_(XXE).md
index 2beba9e..ac349da 100644
--- a/2017/A4-XML_External_Entities_(XXE).md
+++ b/2017/A4-XML_External_Entities_(XXE).md
@@ -13,7 +13,9 @@ exploitability: 2
 prevalence: 2
 detectability: 3
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A4-XML_External_Entities_(XXE).html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A4-XML_External_Entities_(XXE).html
+  - OWASP_Top_Ten_2017/Top_10-2017_A4-XML_External_Entities_(XXE)
 
 ---
 
@@ -79,4 +81,4 @@ Numerous public XXE issues have been discovered, including attacking embedded de
 * [Billion Laughs Attack](https://en.wikipedia.org/wiki/Billion_laughs_attack)<br>
 * [SAML Security XML External Entity Attack](https://secretsofappsecurity.blogspot.tw/2017/01/saml-security-xml-external-entity-attack.html)<br>
 * [Detecting and exploiting XXE in SAML Interfaces](https://web-in-security.blogspot.tw/2014/11/detecting-and-exploiting-xxe-in-saml.html)<br>
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A5-Broken_Access_Control.md b/2017/A5-Broken_Access_Control.md
index eab3d22..55a3702 100644
--- a/2017/A5-Broken_Access_Control.md
+++ b/2017/A5-Broken_Access_Control.md
@@ -13,7 +13,9 @@ exploitability: 2
 prevalence: 2
 detectability: 2
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A5-Broken_Access_Control.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A5-Broken_Access_Control.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A5-Broken_Access_Control
 
 ---
 
@@ -76,4 +78,4 @@ If an unauthenticated user can access either page, it’s a flaw. If a non-admin
 * [CWE-285: Improper Authorization](https://cwe.mitre.org/data/definitions/285.html)<br>
 * [CWE-639: Authorization Bypass Through User-Controlled Key](https://cwe.mitre.org/data/definitions/639.html)<br>
 * [PortSwigger: Exploiting CORS misconfiguration](https://portswigger.net/blog/exploiting-cors-misconfigurations-for-bitcoins-and-bounties)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A6-Security_Misconfiguration.md b/2017/A6-Security_Misconfiguration.md
index 94007a7..69966d7 100644
--- a/2017/A6-Security_Misconfiguration.md
+++ b/2017/A6-Security_Misconfiguration.md
@@ -13,7 +13,9 @@ exploitability: 3
 prevalence: 3
 detectability: 3
 technical: 2
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A6-Security_Misconfiguration.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A6-Security_Misconfiguration.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A6-Security_Misconfiguration
 
 ---
 
@@ -68,4 +70,4 @@ For additional requirements in this area, see the [Application Security Verifica
 * [CWE-388: Error Handling](https://cwe.mitre.org/data/definitions/388.html)<br>
 * [CIS Security Configuration Guides/Benchmarks](https://www.cisecurity.org/cis-benchmarks/)<br>
 * [Amazon S3 Bucket Discovery and Enumeration](https://blog.websecurify.com/2017/10/aws-s3-bucket-discovery.html)<br>
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A7-Cross-Site_Scripting_(XSS).md b/2017/A7-Cross-Site_Scripting_(XSS).md
index cfb3859..7cd4306 100644
--- a/2017/A7-Cross-Site_Scripting_(XSS).md
+++ b/2017/A7-Cross-Site_Scripting_(XSS).md
@@ -13,7 +13,9 @@ exploitability: 3
 prevalence: 3
 detectability: 3
 technical: 2
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A7-Cross-Site_Scripting_(XSS).html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A7-Cross-Site_Scripting_(XSS).html
+  - OWASP_Top_Ten_2017/Top_10-2017_A7-Cross-Site_Scripting_(XSS)
 
 ---
 
@@ -72,4 +74,4 @@ This attack causes the victim’s session ID to be sent to the attacker’s webs
 **External**<br>
 * [CWE-79: Improper neutralization of user supplied input](https://cwe.mitre.org/data/definitions/79.html)<br>
 * [PortSwigger: Client-side template injection](https://portswigger.net/kb/issues/00200308_clientsidetemplateinjection)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A8-Insecure_Deserialization.md b/2017/A8-Insecure_Deserialization.md
index 574abed..ed23c12 100644
--- a/2017/A8-Insecure_Deserialization.md
+++ b/2017/A8-Insecure_Deserialization.md
@@ -13,7 +13,9 @@ exploitability: 1
 prevalence: 2
 detectability: 2
 technical: 3
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A8-Insecure_Deserialization.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A8-Insecure_Deserialization.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A8-Insecure_Deserialization
 
 ---
 
@@ -75,4 +77,4 @@ An attacker changes the serialized object to give themselves admin privileges:<b
 * [CWE-502: eserialization of Untrusted Data](https://cwe.mitre.org/data/definitions/502.html)<br>
 * [Java Unmarshaller Security](https://github.com/mbechler/marshalsec)<br>
 * [OWASP AppSec Cali 2015: Marshalling Pickles](http://frohoff.github.io/appseccali-marshalling-pickles/)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/A9-Using_Components_with_Known_Vulnerabilities.md b/2017/A9-Using_Components_with_Known_Vulnerabilities.md
index 3e8756c..8542849 100644
--- a/2017/A9-Using_Components_with_Known_Vulnerabilities.md
+++ b/2017/A9-Using_Components_with_Known_Vulnerabilities.md
@@ -13,7 +13,9 @@ exploitability: 2
 prevalence: 3
 detectability: 2
 technical: 2
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_A9-Using_Components_with_Known_Vulnerabilities.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_A9-Using_Components_with_Known_Vulnerabilities.html
+  - OWASP_Top_Ten_2017/Top_10-2017_A9-Using_Components_with_Known_Vulnerabilities
 
 ---
 
@@ -67,4 +69,4 @@ There are automated tools to help attackers find unpatched or misconfigured syst
 * [Retire.js for detecting known vulnerable JavaScript libraries](https://github.com/retirejs/retire.js/)<br>
 * [Node Libraries Security Advisories](https://nodesecurity.io/advisories)<br>
 * [Ruby Libraries Security Advisory Database and Tools](https://rubysec.com/)
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/2017/Table_of_Contents.md b/2017/Table_of_Contents.md
index 5345e1b..e43bcb6 100644
--- a/2017/Table_of_Contents.md
+++ b/2017/Table_of_Contents.md
@@ -10,7 +10,9 @@ author:
 contributors:
 tags: OWASP Top Ten 2017, Top Ten
 permalink: 2017/
-redirect_from: OWASP_Top_Ten_2017/Top_10-2017_Table_of_Contents.html
+redirect_from: 
+  - OWASP_Top_Ten_2017/Top_10-2017_Table_of_Contents.html
+  - OWASP_Top_Ten_2017/
 
 ---
 {%- include t10_subsection_begin.html -%}
@@ -51,4 +53,4 @@ OWASP is not affiliated with any technology company, although we support the inf
 The OWASP Foundation is the non-profit entity that ensures the project's long-term success. Almost everyone associated with OWASP is a volunteer, including the OWASP board, chapter leaders, project leaders, and project members. We support innovative security research with grants and infrastructure.
 <br><br>
 Come join us!
-{% include t10_subsection_end.html %}
\ No newline at end of file
+{% include t10_subsection_end.html %}
diff --git a/index.md b/index.md
index 9c75617..2bac948 100644
--- a/index.md
+++ b/index.md
@@ -18,13 +18,13 @@ Companies should adopt this document and start the process of ensuring that thei
 
 ## Top 10 Web Application Security Risks
 
-1. [**Injection**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A1-Injection). Injection flaws, such as SQL, NoSQL, OS, and LDAP injection, occur when untrusted data is sent to an interpreter as part of a command or query. The attacker's hostile data can trick the interpreter into executing unintended commands or accessing data without proper authorization.
-2. [**Broken Authentication**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication). Application functions related to authentication and session management are often implemented incorrectly, allowing attackers to compromise passwords, keys, or session tokens, or to exploit other implementation flaws to assume other users' identities temporarily or permanently.
-3. [**Sensitive Data Exposure**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A3-Sensitive_Data_Exposure). Many web applications and APIs do not properly protect sensitive data, such as financial, healthcare, and PII. Attackers may steal or modify such weakly protected data to conduct credit card fraud, identity theft, or other crimes. Sensitive data may be compromised without extra protection, such as encryption at rest or in transit, and requires special precautions when exchanged with the browser.
-4. [**XML External Entities (XXE)**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A4-XML_External_Entities_(XXE)). Many older or poorly configured XML processors evaluate external entity references within XML documents. External entities can be used to disclose internal files using the file URI handler, internal file shares, internal port scanning, remote code execution, and denial of service attacks.
-5. [**Broken Access Control**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A5-Broken_Access_Control). Restrictions on what authenticated users are allowed to do are often not properly enforced. Attackers can exploit these flaws to access unauthorized functionality and/or data, such as access other users' accounts, view sensitive files, modify other users' data, change access rights, etc.
-6. [**Security Misconfiguration**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A6-Security_Misconfiguration). Security misconfiguration is the most commonly seen issue. This is commonly a result of insecure default configurations, incomplete or ad hoc configurations, open cloud storage, misconfigured HTTP headers, and verbose error messages containing sensitive information. Not only must all operating systems, frameworks, libraries, and applications be securely configured, but they must be patched/upgraded in a timely fashion.
-7. [**Cross-Site Scripting XSS**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A7-Cross-Site_Scripting_(XSS)). XSS flaws occur whenever an application includes untrusted data in a new web page without proper validation or escaping, or updates an existing web page with user-supplied data using a browser API that can create HTML or JavaScript. XSS allows attackers to execute scripts in the victim's browser which can hijack user sessions, deface web sites, or redirect the user to malicious sites.
-8. [**Insecure Deserialization**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A8-Insecure_Deserialization). Insecure deserialization often leads to remote code execution. Even if deserialization flaws do not result in remote code execution, they can be used to perform attacks, including replay attacks, injection attacks, and privilege escalation attacks.
-9. [**Using Components with Known Vulnerabilities**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A9-Using_Components_with_Known_Vulnerabilities). Components, such as libraries, frameworks, and other software modules, run with the same privileges as the application. If a vulnerable component is exploited, such an attack can facilitate serious data loss or server takeover. Applications and APIs using components with known vulnerabilities may undermine application defenses and enable various attacks and impacts.
-10. [**Insufficient Logging & Monitoring**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A10-Insufficient_Logging%252526Monitoring). Insufficient logging and monitoring, coupled with missing or ineffective integration with incident response, allows attackers to further attack systems, maintain persistence, pivot to more systems, and tamper, extract, or destroy data. Most breach studies show time to detect a breach is over 200 days, typically detected by external parties rather than internal processes or monitoring.
+1. [**Injection**](2017/A1-Injection). Injection flaws, such as SQL, NoSQL, OS, and LDAP injection, occur when untrusted data is sent to an interpreter as part of a command or query. The attacker's hostile data can trick the interpreter into executing unintended commands or accessing data without proper authorization.
+2. [**Broken Authentication**](2017/A2-Broken_Authentication). Application functions related to authentication and session management are often implemented incorrectly, allowing attackers to compromise passwords, keys, or session tokens, or to exploit other implementation flaws to assume other users' identities temporarily or permanently.
+3. [**Sensitive Data Exposure**](2017/A3-Sensitive_Data_Exposure). Many web applications and APIs do not properly protect sensitive data, such as financial, healthcare, and PII. Attackers may steal or modify such weakly protected data to conduct credit card fraud, identity theft, or other crimes. Sensitive data may be compromised without extra protection, such as encryption at rest or in transit, and requires special precautions when exchanged with the browser.
+4. [**XML External Entities (XXE)**](2017/A4-XML_External_Entities_(XXE)). Many older or poorly configured XML processors evaluate external entity references within XML documents. External entities can be used to disclose internal files using the file URI handler, internal file shares, internal port scanning, remote code execution, and denial of service attacks.
+5. [**Broken Access Control**](2017/A5-Broken_Access_Control). Restrictions on what authenticated users are allowed to do are often not properly enforced. Attackers can exploit these flaws to access unauthorized functionality and/or data, such as access other users' accounts, view sensitive files, modify other users' data, change access rights, etc.
+6. [**Security Misconfiguration**](2017/A6-Security_Misconfiguration). Security misconfiguration is the most commonly seen issue. This is commonly a result of insecure default configurations, incomplete or ad hoc configurations, open cloud storage, misconfigured HTTP headers, and verbose error messages containing sensitive information. Not only must all operating systems, frameworks, libraries, and applications be securely configured, but they must be patched/upgraded in a timely fashion.
+7. [**Cross-Site Scripting XSS**](2017/A7-Cross-Site_Scripting_(XSS)). XSS flaws occur whenever an application includes untrusted data in a new web page without proper validation or escaping, or updates an existing web page with user-supplied data using a browser API that can create HTML or JavaScript. XSS allows attackers to execute scripts in the victim's browser which can hijack user sessions, deface web sites, or redirect the user to malicious sites.
+8. [**Insecure Deserialization**](2017/A8-Insecure_Deserialization). Insecure deserialization often leads to remote code execution. Even if deserialization flaws do not result in remote code execution, they can be used to perform attacks, including replay attacks, injection attacks, and privilege escalation attacks.
+9. [**Using Components with Known Vulnerabilities**](2017/A9-Using_Components_with_Known_Vulnerabilities). Components, such as libraries, frameworks, and other software modules, run with the same privileges as the application. If a vulnerable component is exploited, such an attack can facilitate serious data loss or server takeover. Applications and APIs using components with known vulnerabilities may undermine application defenses and enable various attacks and impacts.
+10. [**Insufficient Logging & Monitoring**](2017/A10-Insufficient_Logging%252526Monitoring). Insufficient logging and monitoring, coupled with missing or ineffective integration with incident response, allows attackers to further attack systems, maintain persistence, pivot to more systems, and tamper, extract, or destroy data. Most breach studies show time to detect a breach is over 200 days, typically detected by external parties rather than internal processes or monitoring.
diff --git a/info.md b/info.md
index 3a77f6b..6a67176 100644
--- a/info.md
+++ b/info.md
@@ -3,7 +3,7 @@
 * <i class="fas fa-book" style="font-size: 1.2em; color:#233e81;"></i><span style="font-size:1.0em;padding-left:12px;">Documentation</span>
 * <i class="fas fa-toolbox" style="font-size: 1.2em; color:#233e81;"></i><span style="font-size:1.0em;padding-left:12px;">Builder</span> 
 * <i class="fas fa-shield-alt" style="font-size: 1.2em; color:#233e81;"></i><span style="font-size:1.0em;padding-left:12px;">Defender</span>
-* [Current Version (2017)](OWASP_Top_Ten_2017)
+* [Current Version (2017)](2017/)
 
 ### Downloads or Social Links
 * [Download](/www-pdf-archive/OWASP_Top_10-2017_%28en%29.pdf.pdf){:target='_blank' rel='noopener'}
```

(You should be able to apply that diff as a patch.)

Redirects are needed from both the .html links and unspecified extension links, as both seem to be used throughout the web content. (I'd suggest that for the future you don't use *.html links.)

**Comment by ThunderSon** [comment]
Should have handled all of the points above.
For the redirections, I used another regex:
find: `redirect_from: (.*?)(\.html)`
replace:
```
redirect_from:
  - $1$2
  - $1
```

File: 2017/A10-Insufficient_Logging%2526Monitoring.md [hunk 1]
```diff
@@ -40,7 +42,7 @@ Insufficient logging, detection, monitoring and active response occurs any time:
 * Penetration testing and scans by [DAST](/www-community/Vulnerability_Scanning_Tools) tools (such as [OWASP ZAP](/www-project-zap)) do not trigger alerts.<br>
 * The application is unable to detect, escalate, or alert for active attacks in real time or near real time.<br>
 
-You are vulnerable to information leakage if you make logging and alerting events visible to a user or an attacker (see [A3:2017-Sensitive Data Exposure](Top_10-2017_A3-Sensitive_Data_Exposure)).
+You are vulnerable to information leakage if you make logging and alerting events visible to a user or an attacker (see [A3:2017-Sensitive Data Exposure](A3-Sensitive_Data_Exposure.md)).
```
Comment (kingthorin) [inline 2017/A10-Insufficient_Logging%2526Monitoring.md:45]: Don't use any extension just `A3-Sensitive_Data_Exposure` should be fine....same throughout.

File: index.md [hunk 2]
```diff
@@ -18,13 +18,13 @@ Companies should adopt this document and start the process of ensuring that thei
...
-10. [**Insufficient Logging & Monitoring**](/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A10-Insufficient_Logging%252526Monitoring). Insufficient logging and monitoring, coupled with missing or ineffective integration with incident response, allows attackers to further attack systems, maintain persistence, pivot to more systems, and tamper, extract, or destroy data. Most breach studies show time to detect a breach is over 200 days, typically detected by external parties rather than internal processes or monitoring.
+1. [**Injection**](/www-project-top-ten/2017/A1-Injection). Injection flaws, such as SQL, NoSQL, OS, and LDAP injection, occur when untrusted data is sent to an interpreter as part of a command or query. The attacker's hostile data can trick the interpreter into executing unintended commands or accessing data without proper authorization.
```
Comment (kingthorin) [inline index.md:21]: You can drop the `/www-project-top-ten` portion. 

`2017/` is the root of the site (for the purposes of these repos/projects).

**Comment by kingthorin** [comment]
Thanks @ThunderSon! I'll re-test shortly.

**Comment by kingthorin** [comment]
@sslHello 

1. I don't seem to be able to resolve my two comments above.
2. Are you good with this?

**Comment by ThunderSon** [comment]
@kingthorin I resolved them.

**Comment by kingthorin** [comment]
Oh may have missed one file?
`_data/i18n/en.yml `: 
  - `baseURL:                        /OWASP_Top_Ten_2017/`

I'm not sure how/were this YAML file is used.

**Comment by ThunderSon** [comment]
I have my doubts on this one. I'll wait till @sslHello is around. The base URL is pointing to the root, and roots usually are mapped to the OWASP root, and not project roots. I'll update however Torsten sees fit.

**Comment by sslHello** [comment]
@kingthorin   
> Oh may have missed one file?
> `_data/i18n/en.yml `:
> 
>     * `baseURL:                        /OWASP_Top_Ten_2017/`
>  
> I'm not sure how/were this YAML file/value is used.
This file is used for internationalisation of headlines and values in tables and boxes that are often used.
We'd like to encouage volunteers  to compile web pages in additional languages, e.g.
* www-project-top-ten/2017/fr/A1_2017-Injection
* www-project-top-ten/2017/de/A1_2017-Injection
Cheers Torsten

**Comment by sslHello** [comment]
Many thanks to @ThunderSon and @kingthorin;
Commit #27 (37014c4) is based on this pull request and should solve issue #5.
.... sorry for the delay... 
Cheers Torsten

**Pull request closed without merging**